playbooks/roles/edxapp/tasks/deploy.yml

```diff
@@ -80,7 +80,7 @@
     repo: "{{ edxapp_theme_source_repo }}"
     version: "{{ edxapp_theme_version }}"
     accept_hostkey: yes
-  when: edxapp_theme_name != ''
+  when: edxapp_theme_name != '' and EDXAPP_ENABLE_COMPREHENSIVE_THEMING == false
```

Comment:
`when: edxapp_theme_name and not EDXAPP_ENABLE_COMPREHENSIVE_THEMING` should work just fine.